packages/nep141-erc20/src/natural-erc20/sendToNear/index.js

```diff
@@ -89,6 +89,58 @@ export function checkStatus (transfer) {
   }
 }
 
+// Recover transfer from a lock tx hash
+// Track a new transfer at the completedStep = LOCK so that it can be minted
+export async function recover (lockTxHash) {
+  const web3 = new Web3(getEthProvider())
+  const receipt = await web3.eth.getTransactionReceipt(lockTxHash)
+  const ethTokenLocker = new web3.eth.Contract(
+    JSON.parse(process.env.ethLockerAbiText),
+    process.env.ethLockerAddress
+  )
+  const [lockedEvent] = await ethTokenLocker.getPastEvents('Locked', {
+    filter: { transactionHash: lockTxHash },
+    fromBlock: receipt.blockNumber
+  })
+  if (lockedEvent.event !== 'Locked') {
+    throw new Error(
+      'Unable to process lock transaction, for event %s, expected %s',
+      lockedEvent.event, 'Locked'
+    )
+  }
+  const erc20Address = lockedEvent.returnValues.token
+  const amount = lockedEvent.returnValues.amount
+  const recipient = lockedEvent.returnValues.accountId
+  const sender = lockedEvent.returnValues.sender
+  const sourceTokenName = await getName(erc20Address)
+  const decimals = await getDecimals(erc20Address)
+  const destinationTokenName = sourceTokenName + 'ⁿ'
```

Comment:
Same comments about DRYing the code and using typedoc comments apply throughout this file